Navigation/include/ActsLUXEPipeline/LUXEMeasurementsCreator.hpp

```diff
@@ -0,0 +1,246 @@
+            // Construct a particle object
+            Acts::Vector4 particlePos = {pos3[0],pos3[1],pos3[2],parameters[Acts::eBoundTime]};
+            ActsFatras::Particle tempParticle =
+                    makeParticle(
+                            stepper.particleHypothesis(state.stepping).absolutePdg(),stepper.direction(state.stepping),
+                            stepper.charge(state.stepping)/parameters[Acts::eBoundQOverP],
+                            particlePos);
+            std::vector<Acts::MaterialSlab> layerMaterials = {makeSiliconSlab()};
+            std::random_device rd;
+            std::mt19937 gen(rd());
+            auto scattering = ActsFatras::GaussianMixtureScattering();
+            ActsFatras::BetheBloch BBProcess;
+            ActsFatras::BetheHeitler BHProcess;
+
+            Acts::BoundVector scatteredParameters = parameters;
+            ActsFatras::Particle beforeParticle = tempParticle;
+            for (auto material:layerMaterials) {
+                float p1 = tempParticle.absoluteMomentum();
+                auto scatter = scattering(gen, material, tempParticle);
+                auto BBLoss = BBProcess(gen, material, tempParticle);
+                auto BHLoss = BHProcess(gen, material, tempParticle);
+                float p2 = tempParticle.absoluteMomentum();
+                if (p2 < p1/2) {
+                    scatteredParameters[Acts::eBoundQOverP] =
+                            beforeParticle.charge()/(0.999*beforeParticle.absoluteMomentum());
+                } else {
+                    scatteredParameters[Acts::eBoundQOverP] =
+                            tempParticle.charge()/tempParticle.absoluteMomentum();
+                }
+            }
+
+            Acts::SquareMatrix2 cov = Acts::SquareMatrix2::Identity();
+            Acts::BoundSquareMatrix resetCov = Acts::BoundSquareMatrix::Identity();
+            scatteredParameters[Acts::eBoundPhi] = tempParticle.phi();
+            scatteredParameters[Acts::eBoundTheta] = tempParticle.theta();
+            stepper.resetState(state.stepping, scatteredParameters, resetCov, surface);
+            SimpleSourceLink simpleSL(loc, cov, geoId, sourceId);
+            Acts::SourceLink sl(simpleSL);
+            LUXEDataContainer::SimMeasurement SM{sl, scatteredParameters, trueVertex, static_cast<int>(sourceId)};
+            result.push_back(SM);
+        }
+    };
+/// Propagate the track create smeared measurements from local coordinates.
+
+    class MeasurementsCreator : public IAlgorithm {
+    public:
+        /// @brief The nested configuration struct
+        struct Config {
+            /// The output collection
+            std::string outputCollection = "Measurements";
+            // detector
+            std::shared_ptr<const Acts::Experimental::Detector> detector;
+            // variable bins magnetic field
+            std::shared_ptr<Acts::InterpolatedBFieldMap<LUXEMagneticField::vGrid>> BFieldPtr;
```

Comment:
Should we make it `Acts::MagneticFieldProvider`? Not every field we are going to use is going to be binned